components/style/values/specified/position.rs

```diff
@@ -107,16 +233,16 @@ impl ToComputedValue for Position {
     #[inline]
     fn to_computed_value(&self, context: &Context) -> computed_position::Position {
         computed_position::Position {
-            horizontal: self.horizontal.to_computed_value(context),
```

Comment:
This shouldn't be too hard to implement now. You can convert each position to a pair of `Option`s for length and percentage, and then construct a calc value out of them. In case the keyword is right or bottom, you make the len/percentage negative and add 100%. For center, you add 50% but don't make it negative.